src/java.base/share/classes/java/net/URL.java

```diff
@@ -170,6 +170,15 @@
  * the recommendations advised in <a
  * href="https://tools.ietf.org/html/rfc3986#section-7">RFC3986,
  * Section 7, Security Considerations</a>.
+ * <p>
+ * All {@code URL} constructors may additionally throw
+ * {@link MalformedURLException} if the underlying {@link URLStreamHandler}
+ * implementation rejects, or is known to reject, any of the parameters.
+ * In particular when the stream handler's {@linkplain
+ * URLStreamHandler#parseURL(URL, String, int, int) parseURL method} is
+ * called during URL construction and it throws {@code IllegalArgumentException}.
+ * However, which checks are performed is implementation dependent, and
+ * callers should not rely on such checks for full URL validation.
```

Comment:
I think the API note  will need a few adjustments. 

The first sentence uses the word "additionally" but the text prior to that isn't about throwing exceptions. I think what you mean is in addition to the reasons specified by the URL constructors, in which case this makes it normative. Maybe you could lend on the text "parsed URL fails to comply ..." or if that is too complicated then just drop this part of the API note.

A second sentence uses "In particular when ..." but I don't think URL specifies that it calls the parseURL method. If the URL constructors did specify this when you could reference it here. If you map to document the mapping of ISE to MalformedURLException then I think the URL constructors will need to document that they delegate to parseURL and throw MalformedURLException if the parsing fails with IAE.

Reply:
Thanks @AlanBateman - I have updated the API note, added an `@implSpec` in two of the constructors to spell out that they are calling the stream handler's parseURL method (the first one says it's equivalent to the second one so I don't think `@implSpec` is needed there), and improved the `@throws MalformedURLException` with more details.
Does the new text match your expectations?